packages/preset-arbitrary/src/index.ts

```diff
@@ -0,0 +1,68 @@
+import type { Preset, Variant } from '@unocss/core'
+import { extractorSplit } from '@unocss/core'
+import { getBracket, h } from '../../preset-mini/src/utils'
+import { extractorSplitArbitrary } from './split-arbitrary'
+
+export interface PresetArbitraryOptions {
+}
+
+export const variantVariables: Variant = {
+  name: 'variables',
+  match(matcher, ctx) {
+    if (!matcher.startsWith('['))
+      return
+
+    const [match, rest] = getBracket(matcher, '[', ']') ?? []
+    if (!(match && rest))
+      return
+
+    let newMatcher: string | undefined
+    for (const separator of ctx.generator.config.separators) {
+      if (rest.startsWith(separator)) {
+        newMatcher = rest.slice(separator.length)
+        break
+      }
+    }
+
+    if (newMatcher == null)
+      return
+
+    const variant = h.bracket(match) ?? ''
+    const useParent = variant.startsWith('@')
+    if (!(useParent || variant.includes('&')))
+      return
+
+    return {
+      matcher: newMatcher,
+      handle(input, next) {
+        const updates = useParent
+          ? {
+              parent: `${input.parent ? `${input.parent} $$ ` : ''}${variant}`,
+            }
+          : {
+              selector: variant.replace(/&/g, input.selector),
+            }
+        return next({
+          ...input,
+          ...updates,
+        })
+      },
+    }
+  },
+  multiPass: true,
+}
+
+export default function arbitraryPreset(options: PresetArbitraryOptions = {}): Preset {
+  return {
+    name: '@unocss/preset-arbitrary',
+    variants: [
+      variantVariables,
+    ],
```

Comment:
IMO `variantVariables` should stay in its original place in preset-mini and this package should be renamed to @unocss/extractor-arbitrary like #2368. 

variantVariables has preset-mini specific code and has nothing to do with extracting. Otherwise we have to move cssProperty rule here too and having preset-mini dependency in an extractor package doesn't make sense:

https://github.com/unocss/unocss/blob/8554bf483da79bd51383a6f87f18160f90386a7d/packages/preset-mini/src/_rules/variables.ts#L28-L33

Reply:
And by doing that we can include arbitrary extractor in preset-mini exctractor and avoid breaking change.